doc/tutorials/contributing-clion.md

```diff
@@ -0,0 +1,183 @@
+
+```sh
+git branch testbranch
+git checkout testbranch
+```
+
+Here we create a new branch called _testbranch_ and switch to it as our
+working branch (of course you can name it as you want). The next step is
+to add all files you've changed and also want in your commit to the _stage_.
+Suppose we've changed how `kdb get` works, we now have to add it to our files
+we want to commit (you can add several files for a commit too):
+
+```sh
+git add ./src/tools/kdb/get.cpp
+```
+
+Now we've staged our modified file for our commit. The final step is to
+actually commit it to your online repository. Therefore type:
+
+```sh
+git commit -m "Added functionallity to kdb get. Closes #1234"
+git push origin testbranch
+```
+
+With this, you've published your changes to your remote branch of your
+repository (Assuming you are fixing an issue with the number "1234", otherwise
+omit this text). The next step is merging this change into the original
+repository.
+
+## Creating a Pull Request
+
+This step is most easily done using a browser. Open the web page of the
+git-repository of libelektra (https://github.com/ElektraInitiative/libelektra)
+and log in to your account if have not already done so. Navigate to "Pull
+requests", there you can find a green button called "New pull request". By
+clicking it
+([shortcut](https://github.com/ElektraInitiative/libelektra/compare)), you
+can now create a pull request referencing your forked repository and the
+branch,the modified code resides in. Click on "compare across forks" so that
+you can find and select your branch. Choose "<username>/libelectra" as the
+head-repository and "testbranch" as the compare-branch. Now the green button
+"Create pull requests" should be enabled. By clicking it you can define the
+title of your pull request and write a description of the work you have done.
+Please read the template in this form and include the information stated there
+if possible. Finally by clicking "Create pull request" you've successfully
+created a pull request to merge your changes into the official repository! Now
+maintainers of libelectra will review your code and, if everything is fine,
+merge your changes into the official repository. Otherwise they'll comment on
+this pull request if further changes are needed. To include additional changes
+in this pull request, just commit new code-changes to the same repository and
+branch you've referenced for this pull request, they will be added
+automatically to it. In any case check the output of the automated tests!
+
+## Hints
+
+- Especially for not that well-versed programmers don't forget that when
+  debugging e.g. `KeySet`, which can contain many `Key`-objects, you can only
+  see that there may be more than one key stored in this set by it's
+  _size_-attribute. Just the foremost `Key` will be visible directly using the
+  debugger.
+- Sometimes CLion does not rebuild changed modules accordingly by just running
```

Comment:
Please add something like:

If you think, that the modules should have been rebuilt (e.g. if running `testmod_abc` doesn't rebuild `abc`), please report the issue under https://issues.libelektra.org, so we can fix the CMake dependencies.